src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorConnectionContext.java

```diff
@@ -37,6 +37,7 @@
   private final Log log;
 
   private long startMonitorTime;
+  private long invalidNodeStartTime;
   private int failureCount;
   private boolean isNodeUnhealthy;
```

Comment:
NIT: private boolean nodeUnhealthy;

the "is" should be only for the getter, not the variable name.